decorator.rb

```diff
@@ -0,0 +1,27 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    name = @nameable.correct_name[0, 10]
+    name.length > 10 ? "#{name}..." : name
+  end
```

Comment:
- [ ] In the trimmerDecorator class makes sure that the output of @nameable.correct_name has a maximum of 10 characters. So please fix it ``@nameable.correct_name[0..9]`` as [0,10] it will return 11 character.